students/WWhite/lesson05/Mailroom.py

```diff
@@ -1,29 +1,46 @@
 #!/usr/bin/env python3
 
 # -------------------------------------#
-# Desc: Mailroom Part 2
+# Desc: Mailroom Part 3
 # Dev: Will White
-# Date: 5/8/2018
+# Date: 6/11/2018
 # ChangeLog: (When,Who,What)
 # -------------------------------------#
-
+import math
 
 dict_donations = {
     "bill gates": [25000.00, 3], "monet holt": [50000.00, 2], "jeff bezos": [123500.09, 3],
     "john wick": [120000.00, 2], "john snow": [10.56, 1]
 }  # List of donors
 
 
-def menu_options():  # Function to run the menu options
-    user_input = input('''
-Please enter a number from the following options:
+def prompt_for_number_in_range(prompt_text, min_limit, max_limit):
+
+    min_limit = float(min_limit)
+    max_limit = float(max_limit)
+
+    while True:
+        try:
+            return_number = input(prompt_text)
+            return_number = float(return_number)
+
+            if min_limit <= return_number <= max_limit:
+                break
+            else:
+                print("Number out of range.")
+        except:
```

Comment:
you should catch specific exception, in this case are you catching `float` conversion? if so your try/except should be around float and catch ValueError